packages/dashboard/src/components/delivery-alert-store.tsx

```diff
@@ -0,0 +1,539 @@
+      } catch (e) {
+        console.error(`Failed to retrieve existing delivery alerts: ${e}`);
+        return;
+      }
+
+      const filteredAlertsMap: Record<string, DeliveryAlertData> = {};
+      const taskIdToAlertsMap: Record<string, DeliveryAlertData> = {};
+      const resolvedTaskIds: string[] = [];
+      for (const alert of deliveryAlerts) {
+        if (alert.action !== 'waiting') {
+          // Keeping track of resolved task delivery alerts to filter any
+          // remaining warning and waiting delivery alerts.
+          if (alert.task_id && alert.tier === 'error') {
+            resolvedTaskIds.push(alert.task_id);
+          }
+          continue;
+        }
+
+        // No task involved, and still waiting for user action. There should not
+        // be any longstanding delivery alerts that appear after a refresh, only
+        // the delivery alerts that are currently present and have not been
+        // responded to.
+        if (!alert.task_id) {
+          filteredAlertsMap[alert.id] = { deliveryAlert: alert, taskState: undefined };
+          continue;
+        }
+
+        // Tasks that have not been encountered before will go into
+        // taskIdToAlertsMap first, as an error will supercede a warning.
+        if (!Object.keys(taskIdToAlertsMap).includes(alert.task_id)) {
+          let state: TaskState | undefined = undefined;
+          try {
+            state = (await rmf.tasksApi.getTaskStateTasksTaskIdStateGet(alert.task_id)).data;
+          } catch {
+            console.error(
+              `Failed to fetch task state for ${alert.task_id} for delivery alert ${alert.id}`,
+            );
+          }
+          taskIdToAlertsMap[alert.task_id] = {
+            deliveryAlert: alert,
+            taskState: state,
+          };
+        } else if (alert.tier === 'error') {
+          let state: TaskState | undefined = undefined;
+          try {
+            state = (await rmf.tasksApi.getTaskStateTasksTaskIdStateGet(alert.task_id)).data;
+          } catch {
+            console.error(
+              `Failed to fetch task state for ${alert.task_id} for delivery alert ${alert.id}`,
+            );
+          }
+
+          taskIdToAlertsMap[alert.task_id] = {
+            deliveryAlert: alert,
+            taskState: state,
+          };
+        }
+      }
+
+      // Move all unresolved and up-to-date task related delivery alerts to the
+      // filtered map.
+      for (const taskId of Object.keys(taskIdToAlertsMap)) {
+        if (resolvedTaskIds.includes(taskId)) {
+          continue;
+        }
+        const alertData = taskIdToAlertsMap[taskId];
+        filteredAlertsMap[alertData.deliveryAlert.id] = alertData;
+      }
+      setAlerts(filteredAlertsMap);
+    })();
+
+    const sub = rmf.deliveryAlertObsStore.subscribe(async (deliveryAlert) => {
+      let state: TaskState | undefined = undefined;
```

Comment:
Same suggestion as above. 
const instead of let and `filterAndPushDeliveryAlert(deliveryAlert, state);` function inside the try.